src/dmd/expressionsem.d

```diff
@@ -2795,6 +2795,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             exp.error("undefined identifier `%s`, did you mean %s `%s`?", exp.ident.toChars(), s2.kind(), s2.toChars());
         else if (const p = Scope.search_correct_C(exp.ident))
             exp.error("undefined identifier `%s`, did you mean `%s`?", exp.ident.toChars(), p);
+        else if (exp.ident == Id.dollar)
+            exp.error("undefined identifier `$`");
```

Comment:
`$` is not really an identifier (I know it is implemented as such) its an operator

Reply:
I think that the error message is fine this way. It behaves as an identifier in this situation.